pkg/services/accesscontrol/filter.go

```diff
@@ -0,0 +1,99 @@
+package accesscontrol
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+var sqlIDAcceptList = map[string]struct{}{}
+
+type SQLDialect interface {
+	DriverName() string
+}
+
+// Filter creates a where clause to restrict the view of a query based on a users permissions
+// Scopes for a certain action will be compared against prefix:id:sqlID where prefix is the scope prefix and sqlID
+// is the id to generate scope from e.g. user.id
+func Filter(ctx context.Context, dialect SQLDialect, sqlID, prefix, action string, user *models.SignedInUser) (string, []interface{}, error) {
+	if _, ok := sqlIDAcceptList[sqlID]; !ok {
+		return "", nil, errors.New("sqlID is not in the accept list")
+	}
+
+	if user.Permissions == nil || user.Permissions[user.OrgId] == nil {
+		return "", nil, errors.New("missing permissions")
+	}
+
+	scopes := user.Permissions[user.OrgId][action]
+	if len(scopes) == 0 {
+		return " 1 = 0", nil, nil
```

Comment:
@sakjur WDYT?

Reply:
:joy: Nice, would `dialect.BooleanStr(false)` also work?

Reply:
Dont know, possibly :P

Reply:
Yeah that seems to work